### PR TITLE
libarchive 3.7.1

### PR DESCRIPTION
**Pull request by branchv**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
rpm2cpio test failure is irrelevant, see regression build against master branch, https://github.com/chenrui333/github-action-test/actions/runs/5706847294/job/15462947198

**Review by chenrui333 — APPROVED** [review]

File: Formula/libarchive.rb [hunk 1]
```diff
@@ -43,8 +42,11 @@ def install
     system "make", "install"
 
     # fixes https://github.com/libarchive/libarchive/issues/1819
-    inreplace lib/"pkgconfig/libarchive.pc", "Libs.private: ", "Libs.private: -liconv " if OS.mac?
-    inreplace lib/"pkgconfig/libarchive.pc", "Requires.private: iconv", ""
+    if OS.mac?
+      inreplace lib/"pkgconfig/libarchive.pc", "Libs.private: ", "Libs.private: -liconv "
+      inreplace lib/"pkgconfig/libarchive.pc", "Requires.private: iconv", ""
+    end
+
```
Comment (SMillerDev) [inline Formula/libarchive.rb:49]: Only one is supposed to be mac only

**Review by SMillerDev — CHANGES_REQUESTED** [review]

File: Formula/libarchive.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/libarchive.rb]: confirmed the linux build patch is not needed, just removed it.

```
runner@fv-az593-769:/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ cat /home/linuxbrew/.linuxbrew/Cellar/libarchive/3.7.1/lib/pkgconfig/libarchive.pc
...
Libs.private: -lexpat -llzma -lzstd -llz4 -lb2 -lbz2 -lz
Requires.private:
```

**Review by SMillerDev — APPROVED** [review]

**Comment by chenrui333** [comment]
> actually, I cannot regression build mpv with libarchive 3.7.1
> 
> ```
> $ brew reinstall -s mpv
> Warning: Treating mpv as a formula. For the cask, use homebrew/cask/mpv
> ==> Fetching mpv
> ==> Downloading https://github.com/mpv-player/mpv/archive/refs/tags/v0.36.0.tar.gz
> ==> Downloading from https://codeload.github.com/mpv-player/mpv/tar.gz/refs/tags/v0.36.0
>     # -=#=-   #     #
> ==> Reinstalling mpv
> ==> meson setup build -Dhtml-build=enabled -Djavascript=enabled -Dlibmpv=true -Dlua=luajit -Dlibarchive=enabled -Duch
> Last 15 lines from /Users/rui/Library/Logs/Homebrew/mpv/01.meson:
> Run-time dependency iconv found: YES
> Run-time dependency mujs found: YES 1.3.3
> Run-time dependency lcms2 found: YES 2.15
> Did not find CMake 'cmake'
> Found CMake: NO
> Run-time dependency libarchive found: NO (tried framework)
> 
> meson.build:630:13: ERROR: Dependency lookup for libarchive with method 'pkgconfig' failed: Could not generate cargs for libarchive:
> Package iconv was not found in the pkg-config search path.
> Perhaps you should add the directory containing `iconv.pc'
> to the PKG_CONFIG_PATH environment variable
> Package 'iconv', required by 'libarchive', not found
> ```

same issue reporting as this one, https://github.com/mpv-player/mpv/issues/11012 

@SMillerDev I think we might still need this patch `inreplace lib/"pkgconfig/libarchive.pc", "Requires.private: iconv", ""` for macos

**Comment by chenrui333** [comment]
> Not for Linux like it is now?

yeah, linux seems fine.

this is on linux

```
runner@fv-az593-769:/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ cat /home/linuxbrew/.linuxbrew/Cellar/libarchive/3.7.1/lib/pkgconfig/libarchive.pc
prefix=/home/linuxbrew/.linuxbrew/Cellar/libarchive/3.7.1
exec_prefix=${prefix}
libdir=/home/linuxbrew/.linuxbrew/Cellar/libarchive/3.7.1/lib
includedir=${prefix}/include

Name: libarchive
Description: library that can create and read several streaming archive formats
Version: 3.7.1
Cflags: -I${includedir}
Cflags.private: -DLIBARCHIVE_STATIC
Libs: -L${libdir} -larchive
Libs.private: -lexpat -llzma -lzstd -llz4 -lb2 -lbz2 -lz
Requires.private:
```

while this is on macos

```
$ cat /opt/homebrew/Cellar/libarchive/3.7.1/lib/pkgconfig/libarchive.pc
prefix=/opt/homebrew/Cellar/libarchive/3.7.1
exec_prefix=${prefix}
libdir=/opt/homebrew/Cellar/libarchive/3.7.1/lib
includedir=${prefix}/include

Name: libarchive
Description: library that can create and read several streaming archive formats
Version: 3.7.1
Cflags: -I${includedir}
Cflags.private: -DLIBARCHIVE_STATIC
Libs: -L${libdir} -larchive
Libs.private: -liconv -lmd -lexpat -llzma -lzstd -llz4 -lb2 -lbz2 -lz
Requires.private: iconv
```

File: Formula/libarchive.rb [hunk 2]
```diff
@@ -44,7 +43,7 @@ def install
 
     # fixes https://github.com/libarchive/libarchive/issues/1819
     inreplace lib/"pkgconfig/libarchive.pc", "Libs.private: ", "Libs.private: -liconv " if OS.mac?
-    inreplace lib/"pkgconfig/libarchive.pc", "Requires.private: iconv", ""
```
Comment (chenrui333) [inline Formula/libarchive.rb:47]: I am gonna add this build patch back for macOS as it is needed.

**Comment by chenrui333** [comment]
I am gonna revert the change now cc @SMillerDev (let me know if you think the other way)

**Comment by chenrui333** [comment]
just some pyqt related failures on linux build

```
brew install --only-dependencies f3d
brew install f3d
brew install --only-dependencies itk
brew install itk
brew install --only-dependencies mxnet
brew install mxnet
brew install --only-dependencies opencv
brew install opencv
brew install --only-dependencies siril
brew install siril
brew install --only-dependencies synfig
brew install synfig
brew install --only-dependencies visp
brew install visp
```

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**